src/Microsoft.Identity.Web.DownstreamRestApi/DownstreamRestApi.cs

```diff
@@ -40,257 +41,124 @@ internal class DownstreamRestApi : IDownstreamRestApi
         }
 
         /// <inheritdoc/>
-        public async Task<HttpResponseMessage> CallRestApiForUserAsync(
-            string serviceName,
-            Action<DownstreamRestApiOptions>? calledDownstreamRestApiOptionsOverride = null,
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public Task<HttpResponseMessage> CallRestApiAsync(
+            string? serviceName,
+            Action<DownstreamRestApiOptions>? downstreamRestApiOptionsOverride = null,
             ClaimsPrincipal? user = null,
             HttpContent? content = null,
             CancellationToken cancellationToken = default)
         {
-            DownstreamRestApiOptions effectiveOptions = MergeOptions(serviceName, calledDownstreamRestApiOptionsOverride);
-
-            if (effectiveOptions.Scopes == null)
-            {
-                throw new ArgumentException(ScopesNotConfiguredInConfigurationOrViaDelegate);
-            }
-
-            string apiUrl = effectiveOptions.GetApiUrl();
-
-            using HttpRequestMessage httpRequestMessage = new HttpRequestMessage(
-                effectiveOptions.HttpMethod,
-                apiUrl);
-            if (content != null)
-            {
-                httpRequestMessage.Content = content;
-            }
-
-            string authorizationHeader = await
-                _authorizationHeaderProvider.CreateAuthorizationHeaderForUserAsync(
-                    effectiveOptions.Scopes,
-                    effectiveOptions,
-                    user,
-                    cancellationToken).ConfigureAwait(false);
+            DownstreamRestApiOptions effectiveOptions = MergeOptions(serviceName, downstreamRestApiOptionsOverride);
+            return CallRestApiInternalAsync(serviceName, effectiveOptions, effectiveOptions.RequestAppToken, content,
+                                            user, cancellationToken);
+        }
 
-            httpRequestMessage.Headers.Add(Authorization, authorizationHeader);
-            effectiveOptions.CustomizeHttpRequestMessage?.Invoke(httpRequestMessage);
-            using HttpClient client = _httpClientFactory.CreateClient(serviceName);
-            return await client.SendAsync(httpRequestMessage, cancellationToken).ConfigureAwait(false);
+        public Task<HttpResponseMessage> CallRestApiAsync(
```

Comment:
add ` [MethodImpl(MethodImplOptions.AggressiveInlining)]` here as well? and there is no xml.

Reply:
Done. Thanks!